src/ui/utils.js

```diff
@@ -478,3 +493,31 @@ function getTableCellAtPosition( position ) {
 
 	return isTableCellSelected ? position.nodeAfter : findAncestor( 'tableCell', position );
 }
+
+// Returns bounding rect for list of rects.
+//
+// @param {Array.<module:utils/dom/rect~Rect>|Array.<*>} list List of `Rect`s or any list to map by `mapFn`.
+// @param {Function} mapFn Mapping function for list elements.
+// @returns {module:utils/dom/rect~Rect}
+function createBoundingRect( list, mapFn ) {
```

Comment:
This method (by name) looks _potentially_ interesting for `Rect` class but not now. As a general rule we tend to not create utils used in one place.

However, this made me think that this method is constructed in a way that cannot be moved easily from this place as it operates on `mapFn` to create rects.

For now, I think it would be cleaner to make this method operate on rects only (thus doing one thing only - as opposed to doing 2 now).

So the desired workflow with this method would look like this:

```js
const ranges = Array.from( editor.model.document.selection.getRanges() );
// getRectForTableCell is current mapFn - might be inlined as well
const rects = ranges.map( getRectForTableCell );

return {
    target: mergeRects( rects ),
    positions: BALLOON_POSITIONS
}
```

Reply:
As we discussed on a call, we can leave this in current form. This avoids multiple iterations over same list.